src/components/shared/index.ts

```diff
@@ -0,0 +1,3 @@
+// export * from './Layout';
+export { Layout } from './Layout';
+// export type { LayoutProps } from './Layout';
```

Comment:
In theory, type-only imports and exports might be useful to you -- depending on your tooling.

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html#type-only-imports-and-export